pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -43,11 +46,48 @@ const (
 
 var runInTestMode bool
 var runInDemoMode bool
+var slowOperationBuckets = []float64{30, 90, 180, 360, 720, 1440}
+var reconcileCounters = prometheus.NewCounterVec(prometheus.CounterOpts{
+	Name: "metal3_reconcile",
```

Comment:
Per general Prometheus guidelines, counters should have a `_total` suffix and duration metrics should have a unit suffix.
`metal3_reconcile` -> `metal3_reconcile_total`.
`metal3_operation_register` -> `metal3_operation_register_duration_seconds`
`metal3_operation_inspect` -> `metal3_operation_inspect_duration_seconds`
etc.

Tracking the count of failed operations (eg `metal3_reconcile_failed_total`) could be interesting too.